packages/vite/src/node/logger.ts

```diff
@@ -39,8 +39,13 @@ function clearScreen() {
 
 export function createLogger(
   level: LogLevel = 'info',
-  allowClearScreen = true
+  allowClearScreen = true,
+  customLogger?: Logger
 ): Logger {
+  if (customLogger) {
+    return customLogger
```

Comment:
**suggestion:** We could use a callback instead and pass the `level` and `allowClearScreen` parameters, so the custom logger can use these

Reply:
I think it's better if the user implements this.